components/filter/Filter.tsx

```diff
@@ -0,0 +1,113 @@
+import {observer} from "mobx-react";
+import React from "react";
+import {ActionMeta} from "react-select";
+
+import {ExploreSelectedFilter, IFiltersByGroupName, PropMap, PropNames} from "../../lib/types";
+
+interface IFilterProps {
+    setFilter: (groupNames: string[], actionMeta: ActionMeta<ExploreSelectedFilter>) => void;
+    selectedFiltersByGroupName: IFiltersByGroupName;
+}
+
+const Filter: React.FunctionComponent<IFilterProps> = observer(props => {
+    return (
+        <div className={'filter'}>
+            {Object.keys(props.selectedFiltersByGroupName).map(
+                (filter, i, filters) => {
+                    const numberOfAttributes = filters.length;
+                    const addAnd =
+                        numberOfAttributes > 1 &&
+                        i < numberOfAttributes - 1 ? (
+                            <span className="logicalAnd">
+                                AND
+                            </span>
+                        ) : null;
+
+                    return (
+                        <span className="attributeGroup">
+                            <span
+                                className="attributeGroupName"
+                                onClick={() => {
+                                    props.setFilter([filter], {
+                                        action: 'clear',
+                                    });
+                                }}
+                            >
+                                {
+                                    PropMap[
+                                        PropNames[
+                                            filter as keyof typeof PropNames
+                                            ]
+                                        ].displayName
+                                }
+                            </span>
+
+                            {props.selectedFiltersByGroupName[filter].map(
+                                (value, i, values) => {
+                                    const numberOfValues =
```

Comment:
I think it would be good to move the logic computing the parens etc into a helper function - would make the body of `Filter` easier to read